lib/pbench/cli/server/pbench_create_user.py

```diff
@@ -83,8 +93,18 @@ def execute(self):
     required=False,
     help="If user already exists and this option is specified, the error will not be propagated",
 )
```

Comment:
I sat here staring at this input box for a while figuring out what I wanted to say. I'm at least half inclined to agree with Webb, drop this option, and simply have the `unittests` script ignore a 403 failure. It's not quite that simple.

- You're not currently returning any indication of the actual HTTP status; but maybe it's OK to simply ignore all errors
- More importantly, though, I'm not entirely sure any of this will even help with my problem. The gold tests capture `stdout` and `stderr` which includes the `Logger` output. So unless the error path has _no_ difference from the success path, the gold unit tests will fail anyway if the sequence in the current run differs from the prototype run.

I think I may need to give up on the idea of making this dynamic within the per-test setup, and fall back to having the `unittests` script create a pre-defined set of users regardless of which users are in the test `.db` state file. (Or maybe ignoring errors is fine after all as long as the order and set of dynamic usernames doesn't vary from run to run...)

Reply:
If I exit the command with http status code, would that solve the problem?

Reply:
> If I exit the command with http status code, would that solve the problem?

Not if we also fix it to return 400 for all errors ;-)

But, actually, I had another thought while writing this that I should have seen a while back... this utility isn't going to help me at all for the unit tests in any case, because we don't actually have a running server. We'd need a utility that constructed a `User` model object and added it to the database directly, like I'm doing for datasets in my `pbench-state-manager` and `pbench-import-datasets` utilities.

This REST-based utility makes sense for a production environment if we really want to be able to manage usersfrom outside of the server instance. I'm not convinced that's a strong user case, since we ought to have a user management console inside the dashboard UI. Regardless, I think we _at least_ (maybe _also_) need a local "setup" utility to create the canned `admin` user on install, and _that_ likely shouldn't rely on a fully configured running server ... so the direct DB access would be better, and is also what I need for the unit test environment.

So with this in mind, I've lost interest in the `--exists-ok` option entirely; I can't use it, as this utility is currently constructed, regardless of return values. So let's remove it. Or re-cast this entire utility to work directly on the DB rather than through REST, which would make it appropriate for server installation and CLI unit tests.

And while a CLI/REST interface may be useful in the long run, I'm actually not seeing a strong use case for it right now at all...

Reply:
> let's [...] re-cast this entire utility to work directly on the DB rather than through REST, which would make it appropriate for server installation and CLI unit tests.

That approach would get my vote for the reasons stated in the last paragraph.

Reply:
I think that make sense, I will make the changes to this cli utility to directly work with the database rather than using  the REST layer since its deployed on the same instance as a server.